### PR TITLE
[ROCKETMQ-278] Add clusterlist cmd by specified cluster

### PR DESCRIPTION
**Pull request by lindzh**
###  What is the purpose of the change 
When using mqadmin command clusterlist, it always display all cluster, I want to display specified cluster. JIRA: https://issues.apache.org/jira/browse/ROCKETMQ-278.
### Brief change log
* add mqadmin clusterlist selected by cluster.

### Verifying this change
* This change is a trivial rework / code cleanup with unit test coverage.
* execute `mqadmin clusterlist -c {slecified cluster}` to verify.

Follow this checklist to help us incorporate your contribution quickly and easily:
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/ROCKETMQ/issues/)  filed for the change (usually before you start working on it). Trivial changes like typos do not require a JIRA issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x]  Format the pull request title like [ROCKETMQ-XXX] Fix UnknownException when host config not exist. Each commit in the pull request should have a meaningful subject line and body.
- [x]  Write a pull request description that is detailed enough to understand the pull request :
  - What is the purpose of the change
  - Brief change log
  - Verifying this change
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/incubator-rocketmq/tree/master/test).
- [x]  Run mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle to make sure basic checks pass. Run mvn clean install -DskipITs to make sure unit-test pass. Run mvn clean test-compile failsafe:integration-test to make sure integration-test pass. 
- [x]  If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by vsair** [comment]
nice!

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java [hunk 1]
```diff
@@ -63,11 +65,36 @@ public RouteInfoManager() {
         this.filterServerTable = new HashMap<String, List<String>>(256);
     }
 
-    public byte[] getAllClusterInfo() {
-        ClusterInfo clusterInfoSerializeWrapper = new ClusterInfo();
-        clusterInfoSerializeWrapper.setBrokerAddrTable(this.brokerAddrTable);
-        clusterInfoSerializeWrapper.setClusterAddrTable(this.clusterAddrTable);
-        return clusterInfoSerializeWrapper.encode();
+    public byte[] getAllClusterInfo(String cluster) {
```
Comment (shroman) [inline namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java]: You change the signature of the public class here.
How about having another method -- `public byte[] getAllClusterInfo()`? And you won't need calling `getAllClusterInfo(null)` then too ;)

**Comment by dongeforever** [comment]
LGTM @zhouxinyu @vongosling

File: client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java [hunk 2]
```diff
@@ -1167,10 +1168,13 @@ public Properties getBrokerConfig(final String addr, final long timeoutMillis)
         throw new MQBrokerException(response.getCode(), response.getRemark());
     }
 
-    public ClusterInfo getBrokerClusterInfo(
+    public ClusterInfo getBrokerClusterInfo(String cluster,
```
Comment (vongosling) [inline client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java]: how to insure the api compatibility

File: namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java (hunk 1 above)
Comment (vongosling) [reply namesrv/src/main/java/org/apache/rocketmq/namesrv/routeinfo/RouteInfoManager.java]: I agree @shroman here

File: tools/src/main/java/org/apache/rocketmq/tools/command/topic/TopicListSubCommand.java [hunk 3]
```diff
@@ -65,7 +65,7 @@ public void execute(final CommandLine commandLine, final Options options,
         try {
             defaultMQAdminExt.start();
             if (commandLine.hasOption('c')) {
-                ClusterInfo clusterInfo = defaultMQAdminExt.examineBrokerClusterInfo();
+                ClusterInfo clusterInfo = defaultMQAdminExt.examineBrokerClusterInfo(null);
```
Comment (vongosling) [inline tools/src/main/java/org/apache/rocketmq/tools/command/topic/TopicListSubCommand.java]: null as parameter is not a good practice

Comment (lindzh) [reply tools/src/main/java/org/apache/rocketmq/tools/command/topic/TopicListSubCommand.java]: Thanks,I will change it in the next commit.

**Comment by vongosling** [comment]
@lindzh would you please resolve the conflicts, I will merge it.

**Comment by lindzh** [comment]
@vongosling conflicts has been resolved.

**Comment by vongosling** [comment]
Too much merge request commits, could you mind using git pull --rebase to polish your pr ?

**Comment by vongosling** [comment]
@lindzh Thanks, I will pick up the real codes to merge into the develop branch. Expect you to keep your eyes on the RocketMQ community :-)

**Pull request closed without merging**